src/core/index.js

```diff
@@ -43,7 +43,7 @@ class IPFS extends EventEmitter {
 
     if (typeof options.repo === 'string' ||
         options.repo === undefined) {
-      this._repo = defaultRepo(options.repo)
+      this._repo = defaultRepo(options.repo, options.repoPath)
```

Comment:
This is a typo, right? In the examples, you set `options.repoOptions`.

If you are going to change the API by adding a new option, would you please document it with [the rest of the constructor options in the README](https://github.com/ipfs/js-ipfs/blob/master/README.md#ipfs-constructor)?

Personally, I’m also a little unsure about the value of this. It’s basically just replacing:

```js
new IPFS({
  repo: new Repo('/some/path', {options}
})
```

with

```js
new IPFS({
  repo: '/some/path',
  repoOptions: {options}
})
```

You don’t have to explicitly import `ipfs-repo` anymore to get a fully customizable repo, but the whole behavior is more complicated as a result (`repoOptions` won’t do anything if you don’t also set `repo` or if you set `repo` to an actual repo instance). There are also now multiple ways to do the same repo customization. I think this makes the API a little less clear.

That said, I’m more here to focus on docs, so I’ll try not to push or say any more on this.

Reply:
This is a really good point and I agree. The system has a lot of complexity, so avoiding more options in the API is a good thing in my opinion. I think it also helps delineate using the default repo, versus a custom repo, more clearly.